### PR TITLE
naive mysql ping timeout

### PR DESCRIPTION
**Pull request by takhlaq**

### CODE REVIEW

File: sapphire/mysqlConnector/Connection.cpp [hunk 1]
```diff
@@ -222,6 +234,47 @@ MYSQL* Mysql::Connection::getRawCon()
...
+
+      // todo: probably use a config value for ping timeout and use that instead of hardcoded 30sec leeway
+      int seconds = atoi( row[1] );
+      seconds = seconds - 30 < 1 ? 30 : seconds - 30;
+
+      timeout = std::chrono::seconds( seconds );
+      mysql_free_result( pRes );
+   }
+
+   while( true )
```
Comment (ShelbyZ) [inline sapphire/mysqlConnector/Connection.cpp:265]: Give that gremlin some space to live in:

```cpp
while ( true )
```

File: sapphire/mysqlConnector/Connection.cpp [hunk 2]
```diff
@@ -222,6 +234,47 @@ MYSQL* Mysql::Connection::getRawCon()
    return m_pRawCon;
 }
 
+void Mysql::Connection::pingLoop()
+{
+   // todo: this sleep is so dumb, initialise the loop somewhere outside the constructor
+   std::this_thread::sleep_for( std::chrono::seconds( 5 ) );
+   std::chrono::seconds timeout;
```
Comment (ShelbyZ) [inline sapphire/mysqlConnector/Connection.cpp:241]: Why not extract out below (237-263) into a method called getTimeout() and have it return std::chrono:seconds or btter yet a std::chrono::duration.  Have pingLoop() take the timeout and call it beforehand or move timeout a member variable.

Comment (takhlaq) [reply sapphire/mysqlConnector/Connection.cpp:241]: das tru

File: sapphire/mysqlConnector/Connection.cpp (hunk 1 above)
Comment (takhlaq) [reply sapphire/mysqlConnector/Connection.cpp:265]: nop sapphire doesnt do space between control statement and parentheses

**Comment by SapphireMordred** [comment]
I did implement this differently.

**Pull request closed without merging**